pallets/ocex/src/lib.rs

```diff
@@ -383,12 +383,16 @@ pub mod pallet {
 				signature.verify(bytes.as_slice(), &enclave),
 				Error::<T>::EnclaveSignatureVerificationFailed
 			);
-			let mut withdrawal_map: BoundedBTreeMap<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit> = BoundedBTreeMap::try_from(snapshot.withdrawals.clone()).expect("Bounded BTreeMap could not be created");
+			let mut withdrawal_map: BoundedBTreeMap<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit> = BoundedBTreeMap::new();
+			if let Ok(value) = BoundedBTreeMap::<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit>::try_from(snapshot.withdrawals.clone()){
+				withdrawal_map = value;
+			}
+			else {
+				return Err(Error::<T>::StorageOverflow.into());
+			}
```

Comment:
Can this be simplified like:
```rust 
let mut withdrawal_map: BoundedBTreeMap<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit> = 
    BoundedBTreeMap::try_from(snapshot.withdrawals.clone()).ok_or(Error::<T>::StorageOverflow.into())?;

Reply:
Let me try this

Reply:
I'm getting errors while trying to do this

Reply:
@serhii-temchenko So `ok_or` here is returning the `Error::<T>::StorageOverflow.into()` and storing into withdrawal_map, when we have defined the type to be BoundedBTreeMap and it cannot store the error type, so this code will not compile.

Reply:
How you are building this? I have issue with build conflicting version of `syn` crate in this branch.

Reply:
So I've not updated the Cargo.toml with new dependencies and that is why it is not compiling for you

Reply:
I will do that as soon as I update the upstream branch of Polkadex primitives

Reply:
This is done ✅